### PR TITLE
refactor(overlay): reduce paint/render work when opening a modal

### LINKED ISSUE #1910: [Bug]: Picker menu is very slow to open
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Picker

### Expected behavior

When you click on a picker button, the menu opens pretty immediately.

### Actual behavior

With the latest release it's taking around a second to open.

### Screenshots

_No response_

### What browsers are you seeing the problem in?

_No response_

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/picker/
2. Click on a picker button (and observer the time it takes)
3. Repeat step 2 and notice that it's not faster on subsequent attempts.

### Sample code that illustrates the problem

_No response_

### Logs taken while reproducing problem

_No response_

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
- correct application of theme and remove body margin in benchmarking
- expand `<sp-split-button>` benchmark to include 5 open/close actions on one element rather than init on 100
- update `<sp-picker>` and derivatives to gather menu items on a document fragment rather than in the DOM before projecting into an overlay
- reorder and conditionalize work within the overlay process to reduce unneeded work
- update CSS to reduce paints when opening overlay

We get a small, but clear, perf benefit in our automated benchmarking. Locally, under stress, I'm seeing even better perf benefits to this change. On the order of 2 - 3 frames (32 - 48ms) during open or close actions, and I expect to see that further scale up as this is leveraged in busier and busier code paths.

## Related issue(s)

- fixes #1910

## Motivation and context
Faster is better!

## How has this been tested?

-   [ ] _Test case 1_
    1. Go [here](https://modal-overay-refactor--spectrum-web-components.netlify.app/)
    2. Open the "Performance" tab in DevTools
    3. Start recording
    4. Open and close a theme Picker 2 or 3 times
-   [ ] _Test case 2_
    1. Go [here](https://opensource.adobe.com/spectrum-web-components/)
    2. Open the "Performance" tab in DevTools
    3. Start recording
    4. Open and close a theme Picker 2 or 3 times
-   [ ] _Test case 3_
    1. Compare the perf results of the above. Generally I get a 4ms, 14ms and 11ms series of tasks in the new code and a 5ms, 40ms, and 15ms series of tasks in the old code.

## Screenshots (if appropriate)
image
vs
image

## Types of changes
-   [x] Refactor

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/picker/src/Picker.ts [hunk 1]
```diff
@@ -265,11 +265,29 @@ export class PickerBase extends SizedMixin(Focusable) {
...
+    private generatePopover(deprecatedMenu: Menu | null): void {
+        if (this.popoverFragment) return;
+
+        this.popoverFragment = document.createDocumentFragment();
+        render(this.renderPopover, this.popoverFragment, { host: this });
+        this.popover = this.popoverFragment.children[0] as Popover;
+        this.optionsMenu = this.popover.children[1] as Menu;
+
+        if (deprecatedMenu) {
+            console.warn(
```
Comment (hunterloftis) [inline packages/picker/src/Picker.ts:279]: 👍 👍

File: packages/picker/src/Picker.ts [hunk 2]
```diff
@@ -412,7 +432,7 @@ export class PickerBase extends SizedMixin(Focusable) {
 
     protected render(): TemplateResult {
         return html`
-            ${this.renderButton} ${this.renderPopover}
+            ${this.renderButton}
```
Comment (hunterloftis) [inline packages/picker/src/Picker.ts:435]: Would it make sense to get rid of this function and rename `renderButton` to `render`?

File: packages/picker/src/Picker.ts [hunk 3]
```diff
@@ -460,6 +480,7 @@ export class PickerBase extends SizedMixin(Focusable) {
     protected updateMenuItems(
         event?: MenuItemAddedOrUpdatedEvent | MenuItemRemovedEvent
     ): void {
+        if (this.open && event && event.type === 'sp-menu-item-removed') return;
```
Comment (hunterloftis) [inline packages/picker/src/Picker.ts:483]:
```suggestion
        if (this.open && event?.type === 'sp-menu-item-removed') return;
```

File: test/benchmark/helpers.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ governing permissions and limitations under the License.
 import { html, LitElement, render, TemplateResult } from 'lit';
 import { customElement, property } from 'lit/decorators.js';
 import '@spectrum-web-components/theme/sp-theme.js';
-import '@spectrum-web-components/theme/scale-medium.js';
+import '@spectrum-web-components/theme/scale-large.js';
```
Comment (hunterloftis) [inline test/benchmark/helpers.ts:16]: Just curious - why switch to large scale for benchmarking?

**Review by hunterloftis — DISMISSED** [review]

File: test/benchmark/helpers.ts (hunk 4 above)
Comment (Westbrook) [reply test/benchmark/helpers.ts:16]: It's not so much a switch as a correction. https://github.com/adobe/spectrum-web-components/pull/1971/files#diff-737633c0f2c82bcb5bc7453285cb9547b8d34ac0a4aae636c7ca02410ebb12d2R126 I could fix this separate from this work, but in debugging I found that they didn't match and wanted them to in order to better understand the test results.

Comment (hunterloftis) [reply test/benchmark/helpers.ts:16]: Gotcha! 👍

**Pull request merged**